riscv/triggers.h

```diff
@@ -224,6 +224,12 @@ class mcontrol_t : public trigger_t {
   bool load = false;
 };
 
+class mcontrol6_t : public mcontrol_t {
```

Comment:
This is suspicious. Inheritance should define an _isa_ relationship. Is an mcontrol6 object also an mcontrol object? No. This violates the Liskov substitution principle.

How about first extracting an `mcontrol_common_t` with the fields and logic common to both, and then adding a new subclass for `mcontrol6_t`?

Reply:
Agree